starlette/requests.py

```diff
@@ -222,39 +222,36 @@ async def stream(self) -> typing.AsyncGenerator[bytes, None]:
         yield b""
 
     async def body(self) -> bytes:
-        if not hasattr(self, "_body"):
-            chunks = []
-            async for chunk in self.stream():
-                chunks.append(chunk)
-            self._body = b"".join(chunks)
-        return self._body
+        if "body" not in self.scope:
+            self.scope["body"] = b"".join([chunk async for chunk in self.stream()])
+        return self.scope["body"]
 
     async def json(self) -> typing.Any:
-        if not hasattr(self, "_json"):
+        if "json" not in self.scope:
```

Comment:
Can we avoid adding the `"json"` and `"form"` cases here, and *only* persist the body?

(It's not terrible if there are some situations where we're parsing it twice, and we could *consider* if we also want to deal with that as well or not as a separate issue)

Reply:
Given I'm not longer on this tech stack y'all might want to consider this optimization as a separate PR given this is continuing a pattern that already existed within the class itself. The only difference here is moving it to the `scope`

Reply:
json, body and form are not consolidated under a single raw bytes stream cache now